src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -840,8 +840,8 @@ public function findDefinition($id)
      */
     public function createService(Definition $definition, $id, $tryProxy = true)
     {
-        if ('Symfony\Component\DependencyInjection\Definition' !== get_class($definition)) {
-            throw new RuntimeException(sprintf('Constructing service "%s" from a %s is not supported at build time.', $id, get_class($definition)));
+        if ($definition instanceof DefinitionDecorator) {
+            throw new RuntimeException(sprintf('Constructing decorated service "%s" from a %s is not supported at build time.', $id, get_class($definition)));
```

Comment:
That's why I added the "parent" word above, which is also the vocabulary used to declare such services